### PR TITLE
Refactored molecule linting

### PR DESCRIPTION
**Pull request by ssbarnea**
Changes configuration of linting to a model that does not require
writing plugins in order to support other linters. 

Mainly the new setup executes a shell command as part of the lint sequence step.

When no command is mentioned, linting will be skipped.

* only one place to configure linting in molecule
* user decides what commands to run (more flexible)
* molecule no longer have knowledge of specific linters

More details regarding for this breaking change can be found on linked ticket.

Fixes: #2293

### CODE REVIEW

File: molecule/test/functional/docker/test_scenarios.py [hunk 1]
```diff
@@ -262,7 +261,7 @@ def test_host_group_vars(scenario_to_test, with_scenario, scenario_name):
     options = {'all': True}
     cmd = sh.molecule.bake('test', **options)
     out = pytest.helpers.run_command(cmd, log=False)
-    out = util.strip_ansi_escape(out.stdout.decode('utf-8'))
+    out = util.strip_ansi_escape(out.stdout)
```
Comment (sshnaidm) [inline molecule/test/functional/docker/test_scenarios.py]: how is that related?

File: docs/configuration.rst [hunk 2]
```diff
@@ -83,18 +83,49 @@ Podman
...
+Molecule was able to use up to three linters and while it was aimed to flexible
+about them, it ended up creating more confusions to the users. We decided to
+maximize flexibility by just calling an external shell command.
+
+.. code-block:: yaml
+
+    lint: |
+        yamllint .
+        ansible-lint
+        flake8
```
Comment (sshnaidm) [inline docs/configuration.rst]: Is "env" supported? Worth to mention what are possible options here.
Also from which directory it's executed.

File: zuul.d/layout.yaml [hunk 3]
```diff
@@ -6,7 +6,7 @@
     vars:
       tox_envlist: py27-ansible28-unit
       tox_environment:
-        PYTEST_REQPASS: 559
+        PYTEST_REQPASS: 472
```
Comment (sshnaidm) [inline zuul.d/layout.yaml]: this is very weird thing to do

File: docs/getting-started.rst [hunk 4]
```diff
@@ -103,9 +103,8 @@ keys represent the high level components that Molecule provides. These are:
 * The :ref:`driver` provider. Molecule uses `Docker`_ by default. Molecule uses
   the driver to delegate the task of creating instances.
 
-* The :ref:`linters` provider. Molecule uses :std:doc:`Yamllint
-  <yamllint:index>` by default to ensure that best practices are encouraged
-  when writing YAML.
+* The :ref:`root_lint` command. Molecule can call external commands to ensure
```
Comment (sshnaidm) [inline docs/getting-started.rst]: > The :ref:`root_lint` command

Above it's just `lint` command

File: molecule/command/lint.py [hunk 5]
```diff
@@ -71,18 +73,21 @@ def execute(self):
...
+        if cmd == 'yamllint':
+            LOG.warning(
+                "Deprecated linter command found, migrate to newer syntax using <url>"
+            )
+            cmd = 'yamllint .'
+
+        try:
+            # import pdb; pdb.set_trace()
+            LOG.info("Executing: %s" % cmd)
+            util.run(cmd, shell=True, universal_newlines=True, check=True)
```
Comment (sshnaidm) [inline molecule/command/lint.py]: maybe worth to support `env` here

File: molecule/command/lint.py [hunk 6]
```diff
@@ -71,18 +73,21 @@ def execute(self):
...
+        # v3 migration code:
+        cmd = self._config.lint
+        if cmd == 'yamllint':
+            LOG.warning(
+                "Deprecated linter command found, migrate to newer syntax using <url>"
+            )
+            cmd = 'yamllint .'
+
+        try:
+            # import pdb; pdb.set_trace()
```
Comment (sshnaidm) [inline molecule/command/lint.py]: leftovers?

File: molecule/command/lint.py (hunk 5 above)
Comment (ssbarnea) [reply molecule/command/lint.py]: Not in this PR. In fact this will reuse env from parent process. I want to avoid extra logic arond env because if needed it means that user would not be able to run the linter w/o molecule.

File: zuul.d/layout.yaml (hunk 3 above)
Comment (ssbarnea) [reply zuul.d/layout.yaml]: I know but it worked fine so far. Even got a new PR from a new user which didn't ask anything about it, he just updated the number because he was adding a new test.

I bet you will see the real benefit as soon we will have something changes on Zuul nodes that triggers some tests to skip. We still have to re-enable podman ones, which are skipped now.

File: docs/getting-started.rst (hunk 4 above)
Comment (ssbarnea) [reply docs/getting-started.rst]: Thanks, fixed.

File: molecule/command/lint.py [hunk 7]
```diff
@@ -23,13 +23,19 @@
...
 
 
 class Lint(base.Base):
     """
-    Lint Command Class.
+    Lint command executes external linters.
+
+    You need to remember to install those linters. For convenience, there is a
+    package extra that installs the most common ones, use it like
+    ``pip install "molecule[extra]"``.
```
Comment (dmsimard) [inline molecule/command/lint.py]: Should this be ``pip install molecule[lint]`` rather than ``pip install molecule[extra]`` ?

Comment (ssbarnea) [reply molecule/command/lint.py]: Indeed.
